lib/Arch/Context.cpp

```diff
@@ -17,19 +17,39 @@ DecodingContext::GetContextValue(const std::string &context_reg) const {
     return res->second;
   }
 
-  LOG(FATAL) << "No context value for " << context_reg
-             << " but it is required for decoding";
+  LOG(FATAL) << "Required context reg value for: " << context_reg;
 }
+
 DecodingContext DecodingContext::PutContextReg(std::string creg,
                                                uint64_t value) const {
   std::unordered_map<std::string, uint64_t> new_value(this->context_value);
   new_value.emplace(creg, value);
   return DecodingContext(std::move(new_value));
 }
 
+void DecodingContext::UpdateContextReg(std::string creg, uint64_t value) {
+  this->context_value[creg] = value;
+}
+
+void DecodingContext::DropReg(const std::string &creg) {
+  this->context_value.erase(creg);
+}
+
+bool DecodingContext::HasContextValue(const std::string &creg) {
+  return this->context_value.find(creg) != this->context_value.end();
+}
+
+
+DecodingContext
+DecodingContext::MakeContextRegNonConstant(const std::string &creg) const {
```

Comment:
This name is too long and also does not really say much about what is happening inside.
`CtxWithoutReg( ... )`? Or `CopyWithoutReg( ... )` - it is a method so reader already knows it operates on context.